src/main/scala/rise/core/TypedDSL.scala

```diff
@@ -279,23 +279,23 @@ object TypedDSL {
     }
   }
 
-  final case class TDSL[+T <: Expr](private val e: T) {
+  final case class ToBeTyped[+T <: Expr](private val e: T) {
     def toExpr: Expr = TDSL.infer(e)
-    def >>=[X <: Expr](f: T => TDSL[X]): TDSL[X] = f(e)
+    def >>=[X <: Expr](f: T => ToBeTyped[X]): ToBeTyped[X] = f(e)
   }
 
-  implicit def typed[T <: Expr](e: T): TDSL[Opaque] = TDSL(Opaque(e)())
+  implicit def isTyped[T <: Expr](e: T): ToBeTyped[Opaque] = ToBeTyped(Opaque(e)())
```

Comment:
or maybe `keepType`/`preserveType`? otherwise it sounds like a predicate.

Reply:
Yes, I like that. I will choose `preserveType`.